specification/dartLangSpec.tex

```diff
@@ -16401,7 +16405,8 @@ \subsection{Imports}
 }
 
 \LMHash{}%
-It is a static warning to import two different libraries with the same name unless their name is the empty string.
+It is a compile-time error to import two different libraries with the same name
```

Comment:
"to import" suggests that this only matters for a single library, not transitively?

Reply:
That is actually the implemented behavior.

This rule would be motivated by the ability to detect that the same library has been imported twice in ways that makes the execution consider them as distinct (all the way from slightly different URIs to an actual copy of the file).

So it's pretty much a useless rule when it is not checked transitively.

Also, it's not likely to matter much to anybody, because the use of `library` at all has had very little support for several years.

So it's hardly worth the trouble to fix it up (and handle breakage wherever it might arise).

Maybe we should stick to the plan and just change the spec to make it an error, such that the spec follows the actual behavior of the analyzer?

Reply:
Created https://github.com/dart-lang/language/issues/1084, suggesting that we delete this rule. It's impressively useless.

Reply:
Resolving this discussion, it's handled in #1084.